webapp/src/oEmbed.jsx

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import {id as pluginId} from './manifest';
+import {getDimensionsForProvider, getOEmbedUrl, getProviderForUrl} from './utils';
+
+export default class OEmbed extends Component {
+    static propTypes = {
+        embed: PropTypes.shape({
+            url: PropTypes.string.isRequired,
+        }),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.oembedUrl = getOEmbedUrl(this.props.embed.url);
+        const provider = getProviderForUrl(this.props.embed.url);
+        this.dimensions = getDimensionsForProvider(provider.provider_name);
+        this.state = {
+            html: null,
+            thumbnailUrl: null,
+        };
+    }
+
+    async componentWillMount() {
+        try {
+            const res = await (await fetch(`/plugins/${pluginId}/`, {
+                method: 'POST',
+                headers: {'Content-Type': 'application/json'},
+                body: this.oembedUrl,
+            })).json();
+
+            if (res.html) {
+                this.setState({html: res.html});
+            } else {
+                this.setState({thumbnailUrl: res.thumbnail_url});
+            }
+        } catch (err) {
+            // eslint-disable-next-line no-console
+            console.error(err);
+        }
+    }
+
+    render() {
+        const {html, thumbnailUrl} = this.state;
+        const style = {
+            width: `${this.dimensions.width}px`,
+            height: `${this.dimensions.height}px`,
```

Comment:
what happens if the height is not set? shouldn't you fallback to normal post render?

Reply:
Both `width` and `height` should be set for every provider, so that case should never happen.